analyses/survival-analysis/survival-analysis_histology.Rmd

```diff
@@ -0,0 +1,202 @@
+---
+title: "Survival analysis by TP53 and telomerase activity"
+output: html_notebook
+---
+
+This notebook will do the following survival analysis
+1. Univariant analysis 
+- TP53 classifier score (as a continuous variable)
+- EXTEND score (as a continuous variable)
+- Cancer group, removing cancer groups with less than 10 samples
+- HGG vs. non-HGG
+
+2. Multivariant analysis
+- TP53 classifier score, EXTEND score and whatever disease label we determine is appropriate
+
+3. Plots
+- Probably a forest plot will be generated to display the information.
+
+## Setup
+
+#### Packages and functions
+
+Read in set up script.
+
+```{r Set up library}
+library(survival)
+library(dplyr)
+library(ggplot2)
+```
+
+#### Set up files and directories
+
+Set up output directories. 
+
+```{r Set up directories}
+data_dir <- file.path("..", "..", "data")
+analysis_dir <- file.path("..", "..", "analyses")
+figure_dir <- file.path("..", "..", "figures")
+
+results_dir <- "results"
+plots_dir <- "plots"
+```
+
+Make output directories.
+
+```{r Make output directories}
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+```
+
+## Declare the direcotry for all the data to be used 
+
+```{r define data path}
+metadata_file <- file.path(data_dir, "pbta-histologies.tsv")
+indep_file <- file.path(data_dir,
+                         "independent-specimens.rnaseq.primary-plus-stranded.tsv")
+tp53_file <- file.path(analysis_dir, 
+                        "tp53_nf1_score/results/tp53_altered_status.tsv")
+# only use stranded since tp53 scores for polyA did not have good accuracy
+tel_stranded_file <- file.path(analysis_dir,
+                         "telomerase-activity-prediction/results/TelomeraseScores_PTBAStranded_FPKM.txt")
+# color palette
+ color_file <- file.path(figure_dir, "palettes/broad_histology_cancer_group_palette.tsv")
+```
+
+## Import the metadata
+
+```{r Read in metadata}
+# read in histologies file
+indep_rna <- readr::read_tsv(indep_file)
+
+# remove cell lines by selecting only solid tissue
+histologies <- readr::read_tsv(metadata_file, guess_max = 10000) %>%
+ dplyr::filter(composition=="Solid Tissue" & tumor_descriptor == "Initial CNS Tumor") %>%
```

Comment:
Can you add a comment about why we're filtering by `tumor_description == "Initial CNS Tumor"` here please? I kind of expect that this is equivalent to the primary list of RNA-seq specimens?

Reply:
Yes - this is equivalent to primary list of RNA-seq specimens. I think this filter is as a safe guard? Need input from @jharenza.

Reply:
Using a `distinct` call now.